scripts/build/git-id

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+# get/save/load git-described version + SHA1 build id, usage: git-id [dir-to-save/read-ids]
+
+unknown_version() {
+    v=0.0.0
+    if [ -d .git ]; then
+        d=-$(date -u --date="$(git log --pretty=format:%aD HEAD | head -1)" +%Y%m%d%H%M)
```

Comment:
I think we could follow the `git describe` format here. That is, have something like:
```
sha=$(git rev-parse --short HEAD)
commits=$(git rev-list --count HEAD)
dirty=$(git diff --quiet || echo "-dirty")
gitversion="$v-$commits-g$sha$dirty"
```

What do you think?

Reply:
> I think we could follow the `git describe` format here. That is, have something like:
> 
> ```
> sha=$(git rev-parse --short HEAD)
> commits=$(git rev-list --count HEAD)
> dirty=$(git diff --quiet || echo "-dirty")
> gitversion="$v-$commits-g$sha$dirty"
> ```
> 
> What do you think?

Yes. That looks a much saner/safer default. The tip-commit date was especially a bad idea, as it can easily cause consecutive 0.0.0-version numbers to decrease in an untagged repo.

Reply:
I switched to that one.